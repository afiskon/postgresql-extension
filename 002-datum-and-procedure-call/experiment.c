#include <postgres.h>
#include <utils/builtins.h>
#include <datatype/timestamp.h>
#include <utils/timestamp.h>

PG_MODULE_MAGIC;
PG_FUNCTION_INFO_V1(timestamp_in_out_test);

Datum
timestamp_in_out_test(PG_FUNCTION_ARGS)
{
	static const char* tstamp_str = "Aug 05 00:00:00 1988";
	Timestamp tstamp;

	tstamp = DatumGetTimestamp(DirectFunctionCall3(timestamp_in,
		CStringGetDatum(tstamp_str),
		ObjectIdGetDatum(InvalidOid),
		Int32GetDatum(-1)));

	tstamp_str = DatumGetCString(DirectFunctionCall1(timestamp_out,
		TimestampGetDatum(tstamp)));

	PG_RETURN_TEXT_P(CStringGetTextDatum(tstamp_str));
}
