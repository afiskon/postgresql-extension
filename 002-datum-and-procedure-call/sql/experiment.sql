CREATE EXTENSION experiment;

SELECT timestamp_in_out_test();

DROP EXTENSION experiment;