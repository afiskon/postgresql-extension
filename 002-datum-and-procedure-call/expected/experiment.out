CREATE EXTENSION experiment;
SELECT timestamp_in_out_test();
  timestamp_in_out_test   
--------------------------
 Fri Aug 05 00:00:00 1988
(1 row)

DROP EXTENSION experiment;
